### PR TITLE
MST-380 Sync with Discovery on Orgs and Programs

### PR DESCRIPTION
**Pull request by schenedx**
Create the management command to sync the organization and programs objects from discovery service
## Description
Create a management command that calls the Discovery service API.
It will populate the Registrar service with new Organizations and new Programs.
For each new organization created, a new OrganizationGroup with ReadOrganizationReports group is also created
For each new program created, a new ProgramOrganizationGroup with ReadOrganizationReports group is also created
@edx/masters-devs-cosmonauts Please review

### CODE REVIEW

File: registrar/apps/core/management/commands/sync_with_discovery.py [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+)
+from registrar.apps.core.permissions import (
+    OrganizationReadReportRole,
+    ProgramReadReportRole,
+)
+
+
+logger = logging.getLogger(__name__)
+
+PROGRAM_TYPES_TO_SYNC = ['micromasters', 'masters', 'professional-certificate', 'microbachelors']
```
Comment (zacharis278) [inline registrar/apps/core/management/commands/sync_with_discovery.py]: Any reason we didn't add xseries to this?

File: registrar/apps/core/management/commands/test/test_sync_with_discovery.py [hunk 2]
```diff
@@ -0,0 +1,352 @@
...
+        )
+        cls.discovery_other_org = cls.discovery_organization_dict(
+            cls.other_org.discovery_uuid,
+            cls.other_org.key,
+            cls.other_org.name,
+        )
+
+    def assert_org_nonexistant(self, expected_uuid):
+        """
+        Make sure the passed in org uuids do not exists in the database
```
Comment (zacharis278) [inline registrar/apps/core/management/commands/test/test_sync_with_discovery.py]: NIT
```suggestion
        Make sure the passed in org uuids do not exist in the database
```

**Review by zacharis278 — APPROVED** [review]
Pending the changes for multiple authoring orgs, this looks good to me

File: registrar/apps/core/management/commands/sync_with_discovery.py (hunk 1 above)
Comment (schenedx) [reply registrar/apps/core/management/commands/sync_with_discovery.py]: My ignorance caused this. Will add.

**Review by zacharis278 — APPROVED** [review]
👍

**Pull request merged**